src/realm/sync/instruction_replication.cpp

```diff
@@ -674,6 +674,17 @@ void SyncReplication::dictionary_erase(const CollectionBase& dict, size_t ndx, M
     }
 }
 
+void SyncReplication::dictionary_clear(const CollectionBase& dict)
+{
+    Replication::set_clear(dict);
+
+    if (select_collection(dict)) {
+        Instruction::Clear instr;
+        populate_path_instr(instr, dict);
+        emit(instr);
+    }
+}
```

Comment:
please add a sync test that checks that a dictionary clear instruction is round tripped correctly

Reply:
I debugged and tested this. https://github.com/realm/realm-core/blob/cf4ef7ef2f6cda59dbf3142ef2a08f720b96707d/test/object-store/sync/app.cpp#L3171 

It seems to me that the clear instruction is round tripped correctly. I have also added a generic listener in order to check if the flags were set, but for all the collection that flag is not set (although the collection itself is cleared). 
I am not sure if this is due to some bug or the CNF of this particular test, because the same thing done with the C-API worked. 

I left the test I added commented in this PR, you can have a look.

Reply:
I think the test just clears the dictionary and makes sure it get a notification for that. Ideally we would have a test with two users: one clears the dictionary and the other receives the correct instruction (and checks the dictionary is empty on their side too)

Reply:
where can I add such a test?

Reply:
flx_sync.cpp would be a good choice. There are examples of two clients exchanging data. You could even adapt one of those tests.

Reply:
I added a test in app.cpp for partition sync.